exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetPushDownFilter.java

```diff
@@ -165,12 +167,32 @@ protected void doOnMatch(RelOptRuleCall call, FilterPrel filter, ProjectPrel pro
       return;
     }
 
-
     RelNode newScan = ScanPrel.create(scan, scan.getTraitSet(), newGroupScan, scan.getRowType());;
 
     if (project != null) {
       newScan = project.copy(project.getTraitSet(), ImmutableList.of(newScan));
     }
+
+    RowsMatch matchAll = RowsMatch.ALL;
+    if (newGroupScan instanceof AbstractParquetGroupScan) {
+      List<RowGroupInfo> rowGroupInfos = ((AbstractParquetGroupScan) newGroupScan).rowGroupInfos;
+      for (RowGroupInfo rowGroup : rowGroupInfos) {
+        if (rowGroup.getRowsMatch() !=  RowsMatch.ALL) {
+          matchAll = RowsMatch.SOME;
+          break;
+        }
+      }
+    } else {
+      matchAll = RowsMatch.SOME;
+    }
+    // If all groups are in RowsMatch.ALL, no need to apply the filter again to their rows => prune the filter
+    if (matchAll == ParquetFilterPredicate.RowsMatch.ALL) {
```

Comment:
Move this inside `if (newGroupScan instanceof AbstractParquetGroupScan) {`.

Reply:
Done